qutebrowser/app.py

```diff
@@ -301,9 +297,32 @@ def process_pos_args(args, via_ipc=False, cwd=None, target_arg=None):
                 message.error("Error in startup argument '{}': {}".format(
                     cmd, e))
             else:
-                background = open_target in ['tab-bg', 'tab-bg-silent']
-                tabbed_browser.tabopen(url, background=background,
-                                       related=False)
+                win_id = open_url(url, target=open_target)
+
+
+def open_url(url, target=None, force_raise=None):
+    """Open an URL in new window/tab
+
+    Args:
+        url: An URL to open
+        target: same as new_instance_open_target (used as a default)
+        force_raise: control target window raising:
+                     * None - obey new_instance_open_target
+                     * True - always raise
+                     * False - never raise
+
+    Return:
+        ID of a window that was used to open URL
+    """
+    target = target or config.val.new_instance_open_target
+    background = target in {'tab-bg', 'tab-bg-silent'}
+    win_id = mainwindow.get_window(True, force_target=target,
```

Comment:
This still needs to pass `via_ipc` from `process_pos_args` correctly, otherwise it breaks opening URLs in an existing instance from the commandline.

Reply:
Maybe it's better to move `via_ipc` logic to `process_pos_args` than? I'm talking about [this](https://github.com/qutebrowser/qutebrowser/blob/master/qutebrowser/mainwindow/mainwindow.py#L58) and especially [this](https://github.com/qutebrowser/qutebrowser/blob/master/qutebrowser/mainwindow/mainwindow.py#L70).

Reply:
There are four places where `process_pos_args` calls `get_window`, so you'd need to do something like:

```python
if via_ipc:
    win_id = mainwindow.get_window(via_ipc, force_tab=True)
else:
    win_id = 0
```

in four places. Not sure that's any better. Maybe there's a way to make the whole logic a bit simpler, but that seems outside of the scope of this PR.

I'm not sure what you mean with the second link exactly.

Reply:
What I've meant by the second link - that logic doesn't seem natural there.

As far as I understand:
* This code is only reachable if user invoked a command via IPC and no URLs were specified prior to it
* Window is raised for `tab` and `tab-bg`(why not for `window` as well?)

That looks like a narrow case, placed where it does not belong...

But you're right - changing such things at the week of planned release is not a good idea. I've added a `via_ipc` arg to `open_url`.

Reply:
Yeah, it's probably time to look at how to simplify that logic, but after v1.0.0. I opened #3079 for it and will continue the discussion there.